source/agora/test/EmptyBlocks.d

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+
+    Contains the simplest empty block creating test
+
+    Copyright:
+        Copyright (c) 2019-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.EmptyBlocks;
+
+version (unittest):
+
+import agora.api.Validator;
+import agora.consensus.data.Transaction;
+import agora.test.Base;
+
+/// EmptyBlocks test
+unittest
+{
+    auto network = makeTestNetwork!TestAPIManager(TestConf.init);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Create blocks for 2 validator cycles and a couple more
+    auto target_height = Height(2 * GenesisValidatorCycle + 2);
+    network.generateBlocks(target_height, true);
+    network.assertSameBlocks(target_height);
```

Comment:
This doesn't check that the block only have a CoinBase though ?

Reply:
We could, if we really want to be explicit. But since we use `no_txs`, externalization of blocks is enough evidence that it works

I'll add it regardless, for clarity.